beacon-chain/rpc/eth/light-client/helpers.go

```diff
@@ -311,3 +317,74 @@ func newLightClientUpdateToJSON(input *v2.LightClientUpdate) *structs.LightClien
 		SignatureSlot:           strconv.FormatUint(uint64(input.SignatureSlot), 10),
 	}
 }
+
+func isSyncCommitteeUpdate(update *ethpbv2.LightClientUpdate) bool {
+	nextSyncCommitteeBranch := make([][]byte, fieldparams.NextSyncCommitteeBranchDepth)
+	return !reflect.DeepEqual(update.NextSyncCommitteeBranch, nextSyncCommitteeBranch)
+}
+
+func isFinalityUpdate(update *ethpbv2.LightClientUpdate) bool {
+	finalityBranch := make([][]byte, finalityBranchNumOfLeaves)
+	return !reflect.DeepEqual(update.FinalityBranch, finalityBranch)
+}
+
+func isBetterUpdate(newUpdate, oldUpdate *ethpbv2.LightClientUpdate) bool {
+	maxActiveParticipants := newUpdate.SyncAggregate.SyncCommitteeBits.Len()
+	newNumActiveParticipants := uint64(0)
+	for i := uint64(0); i < maxActiveParticipants; i++ {
+		if newUpdate.SyncAggregate.SyncCommitteeBits.BitAt(i) {
+			newNumActiveParticipants += 1
+		}
+	}
+	oldNumActiveParticipants := uint64(0)
+	for i := uint64(0); i < oldUpdate.SyncAggregate.SyncCommitteeBits.Len(); i++ {
+		if oldUpdate.SyncAggregate.SyncCommitteeBits.BitAt(i) {
+			oldNumActiveParticipants += 1
+		}
+	}
+	newHasSupermajority := newNumActiveParticipants*3 >= maxActiveParticipants*2
+	oldHasSupermajority := oldNumActiveParticipants*3 >= maxActiveParticipants*2
+
+	if newHasSupermajority != oldHasSupermajority {
+		return newHasSupermajority
+	}
+	if !newHasSupermajority && newNumActiveParticipants != oldNumActiveParticipants {
+		return newNumActiveParticipants > oldNumActiveParticipants
+	}
+
+	// Compare presence of relevant sync committee
+	newHasRelevantSyncCommittee := isSyncCommitteeUpdate(newUpdate) && (slots.ToEpoch(newUpdate.AttestedHeader.Slot) == slots.ToEpoch(newUpdate.SignatureSlot))
+	oldHasRelevantSyncCommittee := isSyncCommitteeUpdate(oldUpdate) && (slots.ToEpoch(oldUpdate.AttestedHeader.Slot) == slots.ToEpoch(oldUpdate.SignatureSlot))
+
+	if newHasRelevantSyncCommittee != oldHasRelevantSyncCommittee {
+		return newHasRelevantSyncCommittee
+	}
+
+	// Compare indication of any finality
+	newHasFinality := isFinalityUpdate(newUpdate)
+	oldHasFinality := isFinalityUpdate(oldUpdate)
+	if newHasFinality != oldHasFinality {
+		return newHasFinality
+	}
+
+	// Compare sync committee finality
+	if newHasFinality {
+		newHasSyncCommitteeFinality := slots.ToEpoch(newUpdate.FinalizedHeader.Slot) == slots.ToEpoch(newUpdate.AttestedHeader.Slot)
+		oldHasSyncCommitteeFinality := slots.ToEpoch(oldUpdate.FinalizedHeader.Slot) == slots.ToEpoch(oldUpdate.AttestedHeader.Slot)
```

Comment:
Same as above `slots.ToEpoch` probably needs to be implemented in `compute_sync_committee_period_at_slot`as per [specs](https://github.com/ethereum/consensus-specs/blob/208da34ac4e75337baf79adebf036ab595e39f15/specs/altair/light-client/sync-protocol.md#compute_sync_committee_period_at_slot)